streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopicManager.java

```diff
@@ -217,15 +216,17 @@ public void makeReady(final Map<String, InternalTopicConfig> topics) {
     /**
      * Check the existing topics to have correct number of partitions; and return the remaining topics that needs to be created
      */
-    private Set<String> validateTopics(final Set<String> topicsToValidate,
-                                       final Map<String, InternalTopicConfig> topicsMap) {
+    private Set<String> validateTopics(final Set<String> topicsToValidate, final Map<String, InternalTopicConfig> topicsMap) {
+        if (!topicsMap.keySet().containsAll(topicsToValidate)) {
```

Comment:
Do we still need to pass in the topicsMap, as we are no longer looping it?

Reply:
Yes we need the InternalTopicConfig object from it :)

Reply:
More specifically we need the `numberOfPartitions` to compare.